lib/cocoapods-core/specification.rb

```diff
@@ -255,7 +255,7 @@ def recursive_subspecs
     #
     # @return   [Specification] the subspec with the given name or self.
     #
-    def subspec_by_name(relative_name, raise_if_missing = true)
+    def subspec_by_name(relative_name, raise_if_missing = true, include_test_specications = false)
```

Comment:
this is needed eventually during integration. Basically, sometimes we want to find a subspec by name when the `resolver.rb` is resolving dependencies, but if you see above in this file `subspec_dependencies` should *not* include test specification as dependencies.

The added flag helps differentiate when we want to filter test specs or not.